external/boost/archive/portable_binary_archive.hpp

```diff
@@ -37,7 +37,8 @@ enum portable_binary_archive_flags {
 //#endif
 
 inline void
-reverse_bytes(char size, char *address){
+reverse_bytes(signed char size, char *address){
+    BOOST_ASSERT(size > 0);

```

Comment:
Replaced assertion with exception. I'm pretty sure that the calling code never sends size<=0, though.